pkg/k8s/scaledobject.go

```diff
@@ -17,26 +19,45 @@ func kedaGVR() schema.GroupVersionResource {
 
 // NewScaledObjectClient returns a new dynamic client capable
 // of interacting with ScaledObjects in a cluster
+// TODO: This seems not to be called anywhere, remove it?
 func NewScaledObjectClient(cl dynamic.Interface) dynamic.NamespaceableResourceInterface {
 	return cl.Resource(kedaGVR())
 }
 
 // DeleteScaledObject deletes a scaled object with the given name
-func DeleteScaledObject(name string, cl dynamic.ResourceInterface) error {
-	return cl.Delete(name, &v1.DeleteOptions{})
+func DeleteScaledObject(ctx context.Context, metaInfo NameNamespaceInfo, cl client.Client) error {
```

Comment:
this new `NameNamespaceInfo` is used only here. can you just pass two `string`s like we do elsewhere? Another option is to use a `metav1.ObjectMeta`, but I think even that is a bit overkill

Reply:
Changed that, but apparently this function is not being used anywhere too